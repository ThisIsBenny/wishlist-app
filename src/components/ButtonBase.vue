<script lang="ts" setup>
defineProps({
  icon: {
    type: Object,
    default: null,
  },
  mode: {
    type: String,
    default: 'secondary',
  },
})
</script>
<template>
  <button
    class="inline-flex w-fit flex-row items-center justify-center rounded-md border-2 py-1 px-2 disabled:cursor-not-allowed disabled:opacity-60"
    :class="{
      'border-0 bg-cyan-600 text-white disabled:bg-cyan-600 disabled:hover:bg-cyan-600 dark:bg-cyan-600 dark:hover:bg-cyan-600 dark:disabled:hover:bg-cyan-600':
        mode === 'primary',
      'border-0 bg-rose-500 text-white disabled:bg-rose-500 disabled:hover:bg-rose-500 dark:bg-rose-500 dark:hover:bg-rose-500 dark:disabled:hover:bg-rose-500':
        mode === 'danger',
      'border-stone-200 text-stone-500 hover:bg-stone-100 disabled:hover:bg-white dark:border-stone-700 dark:text-white/70 dark:hover:bg-stone-700 disabled:hover:dark:bg-stone-900':
        mode === 'secondary',
    }"
  >
    <component
      v-if="icon"
      :is="icon"
      class="mr-1 h-4 w-4 fill-white"
      :class="{
        'fill-stone-500 dark:fill-white/70 ': mode === 'secondary',
      }"
    />
    <span>
      <slot />
    </span>
  </button>
</template>
