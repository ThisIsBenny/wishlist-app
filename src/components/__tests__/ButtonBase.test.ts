import { describe, assert, it, expect } from 'vitest'
import { shallowMount, VueWrapper } from '@vue/test-utils'
import ButtonBase from '../ButtonBase.vue'

describe('component: ButtonBase', () => {
  let wrapper: VueWrapper

  const createComponennt = (propsData: { mode: string }): void => {
    wrapper = shallowMount(ButtonBase, {
      propsData,
      global: {
        renderStubDefaultSlot: true,
      },
    })
  }
  it('shows primary button', () => {
    createComponennt({ mode: 'primary' })
    expect(wrapper.get('button')).toMatchSnapshot()
  })
  it('shows dange button', () => {
    createComponennt({ mode: 'dange' })
    expect(wrapper.get('button')).toMatchSnapshot()
  })
  it('shows secondary button', () => {
    createComponennt({ mode: 'secondary' })
    expect(wrapper.get('button')).toMatchSnapshot()
  })
  it('shows secondary button as default', () => {
    createComponennt({ mode: '' })
    expect(wrapper.get('button')).toMatchSnapshot()
  })
})
