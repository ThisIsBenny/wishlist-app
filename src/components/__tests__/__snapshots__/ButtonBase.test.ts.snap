// Vitest Snapshot v1

exports[`component: ButtonBase > shows dange button 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <button
    class="inline-flex w-fit flex-row items-center justify-center rounded-md border-2 py-1 px-2 disabled:cursor-not-allowed disabled:opacity-60"
  >
    <!--v-if-->
    <span>
      
      
    </span>
  </button>,
}
`;

exports[`component: ButtonBase > shows primary button 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <button
    class="inline-flex w-fit flex-row items-center justify-center rounded-md border-2 py-1 px-2 disabled:cursor-not-allowed disabled:opacity-60 border-0 bg-cyan-600 text-white disabled:bg-cyan-600 disabled:hover:bg-cyan-600 dark:bg-cyan-600 dark:hover:bg-cyan-600 dark:disabled:hover:bg-cyan-600"
  >
    <!--v-if-->
    <span>
      
      
    </span>
  </button>,
}
`;

exports[`component: ButtonBase > shows secondary button 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <button
    class="inline-flex w-fit flex-row items-center justify-center rounded-md border-2 py-1 px-2 disabled:cursor-not-allowed disabled:opacity-60 border-stone-200 text-stone-500 hover:bg-stone-100 disabled:hover:bg-white dark:border-stone-700 dark:text-white/70 dark:hover:bg-stone-700 disabled:hover:dark:bg-stone-900"
  >
    <!--v-if-->
    <span>
      
      
    </span>
  </button>,
}
`;

exports[`component: ButtonBase > shows secondary button as default 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <button
    class="inline-flex w-fit flex-row items-center justify-center rounded-md border-2 py-1 px-2 disabled:cursor-not-allowed disabled:opacity-60"
  >
    <!--v-if-->
    <span>
      
      
    </span>
  </button>,
}
`;
